rosetta/src/CMakeLists.txt

```diff
@@ -141,3 +141,35 @@ if(ROSETTA_PPM_SYCL)
     target_link_libraries(rosetta-sycl PUBLIC IntelSYCL::SYCL_CXX)
   endif()
 endif()
+
+if(ROSETTA_PPM_HIP)
+  add_library(rosetta-hip rosetta-hip.cpp ${_common_sources})
+  set_target_properties(rosetta-hip PROPERTIES FOLDER "Runtime")
+  target_compile_features(rosetta-hip PUBLIC cxx_std_17)
+  target_compile_features(rosetta-hip PUBLIC hip_std_17)
+
+  # Adding the feature cxx_std_17 does not forward to compiling the rosetta-hip
+  # library. Seems to be a CMake bug. CXX_STANDARD_REQUIRED also does not work
+  set_property(TARGET rosetta-hip PROPERTY CXX_STANDARD 17)
+
+  target_compile_definitions(rosetta-hip PUBLIC ROSETTA_PPM_HIP=1)
+  target_include_directories(rosetta-hip
+                             PUBLIC "${CMAKE_SOURCE_DIR}/rosetta/include")
+  if(HIP_FOUND)
```

Comment:
Why protecting this with `HIP_FOUND`? If HIP wasn't found, then `ROSETTA_PPM_HIP` (condition on line 145) should be false as well, no?